src/documents/views.py

```diff
@@ -213,12 +213,29 @@ def get_serializer(self, *args, **kwargs):
         return super().get_serializer(*args, **kwargs)
 
 
-class CorrespondentViewSet(ModelViewSet, PassUserMixin):
+class PermissionsAwareDocumentCountMixin(PassUserMixin):
+    """
+    Mixin to add document count to queryset, permissions-aware if needed
+    """
+
+    def get_queryset(self):
+        count = (
```

Comment:
I don't have a solution (or am 100% positive anyway), but does this end up querying the database twice or more?  Once to get the documents which the user has permission for, then again in the actual view set to get the actual model?

I'd have to figure out the sql that is output here, but I wonder if there's a way.  I'll look into it more tomorrow as well

Reply:
Thats a very good question and Im not sure of the answer (or how best to check).

Appreciate any thoughts!

Reply:
I think it does end up issues multiple queries, but I don't know a way to combine it all into one.

You can turn on debug logging for `django.db.backends` and it will log the SQL statements it is doing, if you want to debug something like this in the future.

Reply:
Thanks for the tip about the logging. Yes I was looking at it yesterday and seems like you basically have to run a couple queries for this. Maybe some day another SQL guru will do a drive-by PR again